### PR TITLE
Add support of baseline shift

### PR DESCRIPTION
**Pull request by VeselovAlex**

### CODE REVIEW

File: compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt [hunk 1]
```diff
@@ -466,6 +466,12 @@ internal data class ComputedStyle(
         }
 
         res.fontSize = fontSize
+
+        baselineShift?.let {
+            val fontMetrics = res.fontMetrics
+            res.baselineShift = it.multiplier * fontMetrics.ascent
```
Comment (igordmn) [inline compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt:472]: Thanks!

https://github.com/JetBrains/compose-jb/issues/1109 still reproduces (Windows 11)

Comment (VeselovAlex) [reply compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt:472]: This seems to be an issue in Skiko or Skia. Line height is properly calculated if line contains at least one character without baseline shift, even if it is a space. However, I think we should merge this as is and perform fixes in Skiko

Comment (igordmn) [reply compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt:472]: > I think we should merge this as is and perform fixes in Skiko

Without proper testing, we can't be sure that value passed to Skia is correct. For example, we fix issue in Skia, but forget to check in Compose. Baseline can be incorrect after that, even if we were sure that we write code correctly.

I suggest to fix the issue in skiko/skia first before merging this. We can't risk before 1.0.

Comment (igordmn) [reply compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt:472]: I checked the code in the test, and it works correctly. It seems it doesn't work, only if we try to shift an entire line.

In that case (partial work), we can merge it.

Comment (igordmn) [reply compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt:472]: I added a comment [here](https://github.com/JetBrains/compose-jb/issues/1109#issuecomment-979264141)

**Review by igordmn — APPROVED** [review]

**Review by olonho — APPROVED** [review]

**Pull request merged**